cuML/src/dbscan/dbscan.h

```diff
@@ -23,32 +23,52 @@ namespace ML {
 
 using namespace Dbscan;
 
+// Default max mem set to a reasonable value for a 16gb card.
+static const size_t DEFAULT_MAX_MEM_BYTES = 13e9;
```

Comment:
Is there a big downside to halving this to support 8gb cards? Do we have 8gb cards for testing?

Reply:
I used 16gb as the middle ground because I believe most users will be on 16gb and 32gb cards. The expense to lowering the memory too much is increased execution time. Users with 8gb cards will still be able to execute the algorithm, but will need to make a slight adjustment to the `max_bytes_per_batch` argument. I expanded the pydocs to explain this in more detail. Hopefully by 0.8 we will have a better mechanism for computing this automatically.

Reply:
I'm ok deferring a reduction to 0.8. It would be great to take an official position like "we support Pascal+ with at least Xgb" and ensure we actually test on that. Probably basing that on automatically checking memory is most interesting.